modules/peers.js

```diff
@@ -544,7 +545,14 @@ shared.getPeer = function (req, cb) {
 };
 
 shared.version = function (req, cb) {
-	return setImmediate(cb, null, {version: library.config.version, build: library.build});
+	try {
+    	var response = library.build ? {build: library} : {commit: git.getLastCommit()};
+        response.version = library.config.version;
+        return setImmediate(cb, null, response);
+	}
+	catch (err) {
+    	return setImmediate(cb, 'Git is not installed or not a git repository');
```

Comment:
So when there is no build and no git / repository - that endpoint will fail entirely and not return `version`? Imo that endpoint should never fail.